db/migrations/20210126150526_update_get_flap_query.sql

```diff
@@ -0,0 +1,187 @@
+-- +goose Up
+
+ALTER TYPE api.flap_bid_snapshot ADD ATTRIBUTE flap_address TEXT;
+
+-- +goose StatementBegin
+CREATE OR REPLACE FUNCTION api.get_flap_with_address(bid_id NUMERIC, flap_address TEXT, block_height BIGINT DEFAULT api.max_block())
+    RETURNS api.flap_bid_snapshot
+AS
+$$
+WITH address_id AS (SELECT id FROM public.addresses WHERE address = get_flap_with_address.flap_address),
+     storage_values AS (
+         SELECT bid_id,
+                guy,
+                tic,
+                "end",
+                lot,
+                bid,
+                created,
+                updated
+         FROM maker.flap
+         WHERE bid_id = get_flap_with_address.bid_id
+		   AND flap.address_id = (SELECT * from address_id)
+           AND block_number <= block_height
+         ORDER BY block_number DESC
+         LIMIT 1
+     ),
+     deal AS (
+         SELECT deal, bid_id
+         FROM maker.deal
+                  LEFT JOIN public.headers ON deal.header_id = headers.id
+         WHERE deal.bid_id = get_flap_with_address.bid_id
+           AND deal.address_id = (SELECT * FROM address_id)
+           AND headers.block_number <= block_height
+         ORDER BY bid_id, block_number DESC
```

Comment:
nbd but prob don't need to order by bid_id? I think there should only be one bid_id in the results set

Reply:
🤔 that was present in the original query so I didn't think too much about it, but I'm pretty sure you're right.